packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -64,6 +65,33 @@ export interface ITree extends IChannel {
 	): TreeView<TRoot>;
 }
 
+/**
+ * This type parameter is covariant from within the package, but has no impact on assignability from outside the package since the type of private members are dropped.
+ * @public
+ */
+export class InPackageTester<T> {
+	private readonly test!: T;
+}
+
+/**
+ * True when use within this package, false from outside it.
+ * @public
+ */
+export type InPackage = InPackageTester<1> extends InPackageTester<2> ? false : true;
+
+/**
+ * Adapt an interface to be "sealed" making it non-constructable and non implementable, but only from outside the package.
+ * @public
+ */
+export type PackageSeal<T> = InPackage extends true
+	? T
+	: T & ErasedType<readonly ["PackageSeal", T]>;
```

Comment:
Here we get to pick how the type looks from inside the package and outside the package. This is branding it, but other things are possible.